src/tir/transforms/unroll_loop.cc

```diff
@@ -165,6 +170,12 @@ class LoopUnroller : public StmtExprMutator {
     // For loop must have a constant integer extent
     CHECK_NE(value, -1) << "loop doesn't have a constant integer extent";
     if (value == 0) return Evaluate(0);
+    if (explicit_unroll_max_extent_ > 0 && value > explicit_unroll_max_extent_ &&
+        explicit_unroll_) {
```

Comment:
Suggest moving `explicit_unroll_` to the first condition. If it is `false`, we will trigger short circuit evaluation and will not evaluate other conditions.